sonarts-core/src/rules/mccabeComplexityRule.ts

```diff
@@ -49,29 +50,24 @@ export class Rule extends tslint.Rules.AbstractRule {
   }
 
   public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
-    return this.applyWithWalker(new Walker(sourceFile, this.getOptions(), this.threshold));
+    return new Visitor(this.getOptions(), this.threshold).visit(sourceFile).getIssues();
   }
 }
 
-class Walker extends tslint.RuleWalker {
-  threshold: number;
-
-  constructor(sourceFile: ts.SourceFile, options: tslint.IOptions, threshold: number) {
-    super(sourceFile, options);
-    this.threshold = threshold;
+class Visitor extends SonarRuleVisitor {
+  constructor(options: tslint.IOptions, private readonly threshold: number) {
+    super(options.ruleName);
   }
 
-  public visitNode(node: ts.Node) {
-    if (is(node, ...FUNCTION_LIKE)) {
-      const functionComplexity = getFunctionComplexityNodes(node as ts.FunctionLikeDeclaration).length;
-      if (functionComplexity > this.threshold) {
-        this.addFailureAtNode(
-          functionLikeMainToken(node as ts.FunctionLikeDeclaration),
-          Rule.message(functionComplexity, this.threshold),
-        );
-      }
+  public visitFunctionLikeDeclaration(node: ts.FunctionLikeDeclaration) {
+    const functionComplexity = getFunctionComplexityNodes(node as ts.FunctionLikeDeclaration).length;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS4325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=sonar-issue&issues=AWCd633F_Q-0dz4JAbfM&open=AWCd633F_Q-0dz4JAbfM)

[ISSUE_KEY]: AWCd633F_Q-0dz4JAbfM